src/main/java/org/threadly/util/debug/Profiler.java

```diff
@@ -231,6 +311,30 @@ public void run() {
             throw ExceptionUtils.makeRuntime(e.getCause());
           }
         }
+        // start or schedule to handle run time limit
+        if (sampleDurationInMillis > 0) {
+          if (executor instanceof SimpleSchedulerInterface) {
```

Comment:
We would need to release a 3.9.0 version if we want to do that in 4.0.0 release.  You think we should keep the complexity for now, or should we release a 3.9.0 to add the deprecation notices?